ui/src/app/views/profile-page/component.spec.tsx

```diff
@@ -0,0 +1,25 @@
+import {userProfileStore} from 'app/utils/navigation';
+import {mount} from 'enzyme';
+import * as React from 'react';
+import {ProfileStubVariables} from 'testing/stubs/profile-service-stub';
+import {ProfilePageReact} from './component';
+
+
+describe('ProfilePageComponent', () => {
+
+  const profile = ProfileStubVariables.PROFILE_STUB;
+
+  const component = () => {
+    return mount(<ProfilePageReact/>);
+  };
+
+  beforeEach(() => {
+    userProfileStore.next({profile});
+  });
+
+  it('should render the profile', () => {
```

Comment:
Would be good if there were test coverage of (1) validation behavior and (2) saving the profile.

Reply:
Talked with @blrubenstein about this.
For validation behavior, we thought that would be functionally just testing the validation library, which is unnecessary.
For saving, that would really just be testing the profile service, which is also tested independently, right?

Really, there's very little logic in this component.

Reply:
Personally I think we're making a mistake by removing all but a smoke test. There is definitely logic in components like this, and in the long term our velocity & stability will suffer if we don't encode core user expectations somewhere.

(That said, I don't think 100% of the existing tests were worth reproducing. More important / fragile are flows that make async calls and/or involve user interactions.)

I'm writing up something more concrete on the topic to share with the team for discussion.

Reply:
I'm good with writing more tests, I just don't know what tests to write. Any suggestions?

Reply:
For a crud-type page component I'd usually go with a happy-path round-trip test:
 1. Render
 2. Find an input component, edit a value
 3. Trigger click on "submit", ensure the expected interface calls are made (looks like 1 call to updateProfile, and 1 call to props.reload)

I'd also scrutinize L83 a bit – there's no comment indicating what impact this has on the component, but if it's important it might be worth encoding in a test (more for documentation than actual logic)

The adding / removing of institutional affiliations would be lower-priority but useful. (And given the way we're using fp.set, we're not getting benefit from Typescript checking to protect against a typo in the field names.)

Finally, much lower priority but the user interaction to refresh the training completion state is also a bit of round-trip logic that could be checked... but since that's not the main purpose of this component that seems less worrisome if it breaks.

Reply:
+1 to what Greg said. RE validation: I agree we don't need exhaustive testing here (i.e. re-encoding the validator configs you set up), but at least one test with an invalid input to verify that validation is wired up and working properly in the component seems like a good idea to me.

Reply:
Better now?

Reply:
I'm happy – that's a nice example of a very concise test with user interaction & async service calls.

Reply:
Thank @blrubenstein, their influence here is significant